bindings/go/oci/spec/descriptor/decode.go

```diff
@@ -0,0 +1,82 @@
+package descriptor
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	descriptor "ocm.software/open-component-model/bindings/go/descriptor/runtime"
+	v2 "ocm.software/open-component-model/bindings/go/descriptor/v2"
+)
+
+// SingleFileDecodeDescriptor decodes a component descriptor from a TAR archive.
+func SingleFileDecodeDescriptor(raw io.Reader, mediaType string) (*descriptor.Descriptor, error) {
+	switch mediaType {
+	case MediaTypeLegacyComponentDescriptorTar,
+		mediaTypeLegacy2ComponentDescriptorTar,
+		mediaTypeLegacy3ComponentDescriptorTar:
+		descriptorStream, err := descriptorFileFromTar(raw)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get component descriptor stream from tar: %w", err)
+		}
+		descriptorYAML, err := io.ReadAll(descriptorStream)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read component descriptor stream from tar: %w", err)
+		}
+		var v2desc v2.Descriptor
+		if err := yaml.UnmarshalStrict(descriptorYAML, &v2desc); err != nil {
+			return nil, fmt.Errorf("unmarshaling component descriptor: %w", err)
+		}
+		desc, err := descriptor.ConvertFromV2(&v2desc)
+		if err != nil {
+			return nil, fmt.Errorf("converting component descriptor: %w", err)
+		}
+		return desc, nil
+	case MediaTypeComponentDescriptorJSON, MediaTypeLegacyComponentDescriptorJSON,
+		MediaTypeComponentDescriptorYAML, MediaTypeLegacyComponentDescriptorYAML:
+		descriptorYAML, err := io.ReadAll(raw)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read component descriptor stream from tar: %w", err)
+		}
+		var v2desc v2.Descriptor
+		if err := yaml.Unmarshal(descriptorYAML, &v2desc); err != nil {
+			return nil, fmt.Errorf("unmarshaling component descriptor: %w", err)
+		}
+		desc, err := descriptor.ConvertFromV2(&v2desc)
+		if err != nil {
+			return nil, fmt.Errorf("converting component descriptor: %w", err)
+		}
+		return desc, nil
+	default:
+		return nil, fmt.Errorf("unsupported descriptor media type %s", mediaType)
+	}
+}
+
+// descriptorFileFromTar reads the component descriptor from a tar.
+// The component is expected to be inside the tar in a file called descriptorFileHeader.
```

Comment:
This comment isn't correct, right? You are looking for LegacyComponentDescriptorTarFileName.